tests/components/husqvarna_automower/test_init.py

```diff
@@ -127,21 +128,44 @@ async def test_websocket_not_available(
     caplog: pytest.LogCaptureFixture,
     freezer: FrozenDateTimeFactory,
 ) -> None:
-    """Test trying reload the websocket."""
-    mock_automower_client.start_listening.side_effect = HusqvarnaWSServerHandshakeError(
-        "Boom"
-    )
-    await setup_integration(hass, mock_config_entry)
-    assert "Failed to connect to websocket. Trying to reconnect: Boom" in caplog.text
-    assert mock_automower_client.auth.websocket_connect.call_count == 1
-    assert mock_automower_client.start_listening.call_count == 1
-    assert mock_config_entry.state is ConfigEntryState.LOADED
-    freezer.tick(timedelta(seconds=2))
-    async_fire_time_changed(hass)
-    await hass.async_block_till_done()
-    assert mock_automower_client.auth.websocket_connect.call_count == 2
-    assert mock_automower_client.start_listening.call_count == 2
-    assert mock_config_entry.state is ConfigEntryState.LOADED
+    """Test trying to reload the websocket."""
+    # Patch DEFAULT_RECONNECT_TIME to 0 for the test
+    with patch(
+        "homeassistant.components.husqvarna_automower.coordinator.DEFAULT_RECONNECT_TIME",
+        new=0,
+    ):
+        # Simulate a WebSocket handshake error
+        mock_automower_client.auth.websocket_connect.side_effect = (
+            HusqvarnaWSServerHandshakeError("Boom")
+        )
+
+        # Setup integration and verify initial log message
+        await setup_integration(hass, mock_config_entry)
+        assert (
+            "Failed to connect to websocket. Trying to reconnect: Boom" in caplog.text
+        )
+
+        # Initial call count and range for reconnection attempts
+        mock_automower_client.reset_mock()
+        recursion_limit = sys.getrecursionlimit()
+        # Perform reconnection attempts
+        for count in range(recursion_limit):
+            await hass.async_block_till_done()
+            assert mock_automower_client.auth.websocket_connect.call_count == count + 1
+            assert mock_config_entry.state is ConfigEntryState.LOADED
+
+        # Simulate a successful connection
+        mock_automower_client.auth.websocket_connect.side_effect = None
+        await hass.async_block_till_done()
+        assert mock_automower_client.start_listening.call_count == 1
+
+        # Simulate a start_listening TimeoutException
+        mock_automower_client.start_listening.side_effect = TimeoutException("Boom")
+        await hass.async_block_till_done()
+        assert (
+            mock_automower_client.auth.websocket_connect.call_count
+            == recursion_limit + 2
+        )
```

Comment:
I'd reset the side effect again here and test that we connect and listen successfully.

Reply:
And maybe finish the test by having another exception and then stopping `hass` with `hass.async_stop` and checking that we don't try to connect again.

Reply:
> I'd reset the side effect again here and test that we connect and listen successfully.

done in https://github.com/home-assistant/core/pull/123085/commits/196bbb684ea1633ee527406decd2e2d43a602f06
We test the successful connection by checking the caplog. Is that okay?

Reply:
> And maybe finish the test by having another exception and then stopping `hass` with `hass.async_stop` and checking that we don't try to connect again.

Done in https://github.com/home-assistant/core/pull/123085/commits/84378003f1c77a5afde77363f4547ee657690f2c